pkg/status/combinedstatus-resolver.go

```diff
@@ -428,6 +436,13 @@ func (c *combinedStatusResolver) ResolutionExists(name string) (string, util.Obj
 	return key.bindingName, key.sourceObjectIdentifier, true
 }
 
+func (c *combinedStatusResolver) ResolutinosForBinding(bdgName string) map[util.ObjectIdentifier]*combinedStatusResolution {
+	c.RLock()
+	defer c.RUnlock()
+
+	return c.bindingNameToResolutions[bdgName]
```

Comment:
`c.bindingNameToResolutions` should only be accessed while holding the lock in `c.RWMutex`.

I think a workable alternative is to redefine this method to return the set of missing StatusCollector names.

Reply:
Done.

Reply:
Sorry, I was not clear enough. `c.bindingNameToResolutions` is a two-layer map. It maps a string to another _map_. `c.RWMutex` applies to both layers. It is not valid to return an inner map after releasing `c.RWMutex`.

The comment on `c.bindingNameToResolutions` does not promise that the inner maps are immutable. And indeed, they are mutated in some places.

Reply:
Indeed. I implemented the suggested 'workable alternative' above.